src/docfx/config/Config.cs

```diff
@@ -96,9 +102,63 @@ private static Config LoadCore(string configPath, CommandLineOptions options)
         private static JObject Expand(JObject config)
         {
             config[ConfigConstants.Content] = ExpandFiles(config[ConfigConstants.Content]);
+            config[ConfigConstants.FileMetadata] = ExpandGlobConfigs(config[ConfigConstants.FileMetadata]);
             return config;
         }
 
+        private static JToken ExpandGlobConfigs(JToken token)
+        {
+            if (token == null)
+                return null;
+            if (token is JObject obj)
+                token = ToGlobConfigs(obj);
+            if (token is JArray arr)
+            {
+                foreach (var item in arr)
+                {
+                    ExpandGlobConfig(item);
+                }
+            }
+            return token;
+        }
+
+        private static void ExpandGlobConfig(JToken item)
+        {
+            if (!(item is JObject obj))
+                throw new Exception($"Expect to be an object: {JsonUtililty.Serialize(item)}");
+            ExpandIncludeExclude(obj);
+        }
+
+        private static JArray ToGlobConfigs(JObject obj)
+        {
+            var result = new JArray();
+
+            foreach (var (key, value) in obj)
+            {
+                result.Add(new JObject
+                {
+                    [ConfigConstants.Include] = ToGlob(key),
```

Comment:
Can we avoid turning folders into glob patterns? It is a lot faster to check if a file inside a folder using starts with. Glob matchers are slower.

Reply:
resolved